File: Dec2020/13Dec/PizzaConfusion.java
/*
Problem Name: Pizza Confusion
Problem Link: https://www.hackerearth.com/practice/algorithms/sorting/bubble-sort/practice-problems/algorithm/pizza-confusion/
*/
import java.io.*;
import java.util.*;
class TestClass {
    public static void main(String args[] ) throws Exception {
        Scanner in = new Scanner(System.in);
        PrintWriter out = new PrintWriter(System.out);
        int n = in.nextInt();
        String ans = in.next();
        int points = in.nextInt();
        for(int i=1;i<n;i++){
            String name = in.next();
            int p = in.nextInt();
            if(p > points){
                points = p;
                ans = name;
            }
            else if(p == points && ans.compareTo(name) > 0){
                ans = name;
            }
        }
        out.println(ans);
        out.flush();
    }
}

File: Dec2020/13Dec/PizzaConfusion.cpp
/*
Problem Name: Pizza Confusion
Problem Link: https://www.hackerearth.com/practice/algorithms/sorting/bubble-sort/practice-problems/algorithm/pizza-confusion/
*/

#include<bits/stdc++.h>
using namespace std;

#define test(t)     int t; cin>>t; while(t--)
#define ll          long long

int main(){
    ios::sync_with_stdio(0);    
    cin.tie(0);             
    ll n, points, maxpoints;
    string name, rest;
    map<int, vector<string>> m;
    cin>>n;
    cin>>name>>points;
    maxpoints = points;
    rest = name;
    for(ll i=1;i<n;++i){
        cin>>name>>points;
        if(points == maxpoints){
            if(name < rest){
                rest = name;
            }
        }
        else if(points > maxpoints){
            rest = name;
            maxpoints = points;
        }
    }
    cout<<rest;
    return 0;
}
